### DISCUSSION TITLE
Dynamic api route property is undefined.

### DISCUSSION THREAD

**Discussion by olaven** [vercel/next.js#13505] [Help]
Thanks for maintaining this useful framework!

# Bug report

## Describe the bug
__update__: This bug only appears in the node-test. It does not appear when fetching from the browser. 

The dynamic query property (`[id].ts`) is undefined.
My file is located in `pages/api/brands/[id].ts`. 
I am reading the id like this: 
```ts
const {
    query: { id },
  } = request
```

## To Reproduce
1. Create a dynamic API endpoint (e.g. `/pages/api/posts/[id].ts`.
2. Write a Jest test that fetches the endpoint.
3. Try to read the id from `request.query`.
4. `id` is undefined

## Expected behavior
I expected the ID to have the value in my url, e.g. `/api/brands/some-id`. 

## System information

- OS: macOs
- Browser: No browser, jest in node environment
- Version of Next.js: ^9.4.2
- Version of Node.js: v13.13.0

__EDIT__: Removed template text

**Comment by timneutkens** [comment]
Can you provide a full github repository?

**Comment by olaven** [comment]
@timneutkens Sure! Check out https://github.com/olaven/error-reproductions/tree/master/nextjs-api-route-error 

`yarn test` -> is undefined 
`yarn dev`/browser -> is defined

**Comment by timneutkens** [comment]
Ran `yarn dev` and went to:
```
http://localhost:3000/api/posts/123
```

The correct `id` is returned there.

Same if you open:

```
http://localhost:3000
```

The output looks correct as well.

Looks like your testing setup is wrong though. Particularly you're using Next.js internals that you shouldn't touch: `import {apiResolver} from 'next/dist/next-server/server/api-utils'`.

Instead you can `next build` and `next start` and then run the tests against `http://localhost:3000` for example.

Going to convert this to a discussion as it's not a bug in Next.js, the `[id]` is provided correctly.
